src/expr/src/scalar/func.rs

```diff
@@ -1448,6 +1448,30 @@ pub trait TimestampLike: chrono::Datelike + chrono::Timelike + for<'a> Into<Datu
         s + ns
     }
 
+    fn extract_millisecond(&self) -> f64 {
+        let s = f64::from(self.second() * 1_000);
+        let ns = f64::from(self.nanosecond()) / 1e6;
+        s + ns
+    }
+
+    fn extract_microsecond(&self) -> f64 {
+        let s = f64::from(self.second() * 1_000_000);
+        let ns = f64::from(self.nanosecond()) / 1e3;
+        s + ns
+    }
+
+    fn extract_millennium(&self) -> f64 {
+        f64::from((self.year() + if self.year() > 0 { 999 } else { -1_000 }) / 1_000)
+    }
+
+    fn extract_century(&self) -> f64 {
+        f64::from((self.year() + if self.year() > 0 { 99 } else { -100 }) / 100)
+    }
+
+    fn extract_decade(&self) -> f64 {
+        f64::from(self.year().div_euclid(10))
```

Comment:
Euclidean division doesn't produce different results than standard division in the provided test cases. Can you let me know why you chose this method, or provide a test case that standard division fails?

Reply:
The current version of Materialize will return an error on the sqllogictest for decade extraction, "unsupported timestamp units 'decade'". PostgreSQL will return the right result, "200 200 199 0 0 -1 -1 -2".
The years obtain from self.year() are [2001, 2000, 1999, 1, 0, -1, -10, -11] in the test.
Had I implemented it as
```
fn extract_decade(&self) -> f64 {
    f64::from(self.year() / 10)
}
```

That is, using normal division, Materialize would return "200 200 199 0 0 0 -1 -1", which is not how postgres handles it.

Reply:
Ah, you are correct!